src/components/TableOfContents/index.tsx

```diff
@@ -7,12 +7,13 @@ import ListView from '#rsu/../v2/View/ListView';
 import styles from './styles.scss';
 
 interface Props<T, K extends string | number>{
-    keySelector: (datum: T) => K;
+    idSelector: (datum: T) => K;
+    keySelector: (datum: T) => K | undefined;
```

Comment:
What is the difference between key and id?

Reply:
Do we even need key?

Reply:
the server api requires it. i don't know why but it is as such right now.

Reply:
Is key for client and id for server? Why do we need both?

Reply:
Each widget option has id and key which is prevelant throughout code, so i think its fine.
Still, I'll quickly add the reason behind having 2 things.

Reply:
Where was the reason added?